src/Microsoft.ML.TimeSeries/STL/MathsTools.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.TimeSeries
+{
+    internal class MathUtility
+    {
+        /// <summary>
+        /// calculate the standard sigmoid function
+        /// </summary>
+        /// <param name="x">the input value</param>
+        public static double Sigmoid(double x)
+        {
+            return 1.0 / (1.0 + Math.Exp(-x));
+        }
```

Comment:
I was going to suggest using this other method we already have for sigmoids:
https://github.com/dotnet/machinelearning/blob/0fcf7ae132a4787cb6df414e4d8b49b35d509ff4/src/Microsoft.ML.Core/Utilities/MathUtils.cs#L482-L489

but then I realized that other method only accepts floats, and not doubles.

On the other hand, notice that there is the method `SigmoidFast` which seems to have some logic to approximate Sigmoid values, and is, I guess, faster. So maybe it's worth a look there to see if using that approximation logic makes your PR more performant 😄

Reply:
But, on the other hand, I'm just realizing that this method is only used by `TrendInfo` class, which as I mention [here](https://github.com/dotnet/machinelearning/pull/5202/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csv#r440494174), I'm still not sure if you're actually using that class or not.

---
In reply to: [440511795](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440511795) [](ancestors = 440511795)

Reply:
the SigmoidFast is really awesome, and I have removed this function as this function is actually unused in the code :)

---
In reply to: [440512783](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440512783) [](ancestors = 440512783,440511795)